internal/command/jsonformat/computed/renderers/string.go

```diff
@@ -0,0 +1,156 @@
+package renderers
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/hashicorp/terraform/internal/command/format"
+	"github.com/hashicorp/terraform/internal/command/jsonformat/collections"
+	"github.com/hashicorp/terraform/internal/command/jsonformat/computed"
+	"github.com/hashicorp/terraform/internal/plans"
+)
+
+type evaluatedString struct {
+	String string
+	Json   interface{}
+
+	IsMultiline bool
+}
+
+func evaluatePrimitiveString(value interface{}) evaluatedString {
+	if value == nil {
+		return evaluatedString{String: "[dark_gray]null[reset]"}
+	}
+
+	str := value.(string)
+
+	if strings.HasPrefix(str, "{") || strings.HasPrefix(str, "[") {
+		var jv interface{}
+		if err := json.Unmarshal([]byte(str), &jv); err == nil {
+			return evaluatedString{
+				String: str,
+				Json:   jv,
+			}
+		}
+	}
+
+	if strings.Contains(str, "\n") {
+		return evaluatedString{
+			String:      strings.TrimSpace(str),
```

Comment:
The trim is surprising to me here. Does that reflect existing diff rendering behaviour?

Reply:
Yeah it does. I think because we don't render multiline strings surrounded by quotation marks you don't see the whitespace anyway, but it's definitely explicitly reflected in diff_test.go that trailing whitespace is removed.